server/src/main/java/org/opensearch/action/support/TransportAction.java

```diff
@@ -88,31 +89,39 @@ public final Task execute(Request request, ActionListener<Response> listener) {
          */
         final Releasable unregisterChildNode = registerChildNode(request.getParentTask());
         final Task task;
+
         try {
             task = taskManager.register("transport", actionName, request);
         } catch (TaskCancelledException e) {
             unregisterChildNode.close();
             throw e;
         }
-        execute(task, request, new ActionListener<Response>() {
-            @Override
-            public void onResponse(Response response) {
-                try {
-                    Releasables.close(unregisterChildNode, () -> taskManager.unregister(task));
-                } finally {
-                    listener.onResponse(response);
+
+        ThreadContext.StoredContext storedContext = taskManager.taskExecutionStarted(task);
```

Comment:
At least, it is a bit cleaner, thanks @tushar-kharbanda72, why do we do that every time instead of doing in once in the `void execute(Task task, Request request, ActionListener<Response> listener)`?

Reply:
The other execute gets called for an already registered task as well. So essentially it's possible task manager getting execute call twice for same task.